source/wp-content/themes/wporg-developer/js/function-reference.js

```diff
@@ -126,4 +126,23 @@ jQuery( function ( $ ) {
 	}
 
 	toggleUsageListInit();
+
+	// Runs before the highlight parsing is run.
+	// `env` is defined here: https://github.com/PrismJS/prism/blob/2815f699970eb8387d741e3ac886845ce5439afb/prism.js#L583-L588
+	Prism.hooks.add( 'before-highlight', function ( env ) {
+		if ( 'php' === env.language && ! env.code.startsWith( '<?' ) ) {
+			env.code = '<? ' + env.code;
+			env.hasAddedTag = true;
+		}
+	} );
+
+	// Runs before `highlightedCode` is set to the `innerHTML` of the container.
+	Prism.hooks.add( 'before-insert', function ( env ) {
+		if ( env.hasAddedTag ) {
+			env.highlightedCode = env.highlightedCode.replace(
+				'<span class="token delimiter important">&lt;?</span> ',
```

Comment:
I think this needs to be anchored to only replace the initial instance of it perhaps? Just to avoid cases where `<?` exists within any code blocks.

Reply:
Turns out this is already a singular replace, as JS `.replace()` uses a regexp, and you have the explicitly state you want a "global" replace.. TIL.